import catchAsyncError from "../middleware/catch.middleware.js";
import Reservation from "../models/reservation.models.js";
import ErrorHandler from "../utils/errorHandler.js";

const reservationSave = catchAsyncError(async (req, res, next) => {
  const { name, lastname, times, table, numberOfPeople, note, status } =
    req.body;

  if (Object.values(req.body).every((value) => !value)) {
    return next(new ErrorHandler("Hepsini doldurunuz", 400));
  }
  const reserver = await Reservation.create({
    name,
    lastname,
    times,
    table,
    numberOfPeople,
    note,
    status,
  });

  res.status(200).json({ reserver });
});

const reservationGet = catchAsyncError(async (req, res, next) => {
  const reserver = await Reservation.find();

  res.status(200).json({ reserver });
});

const reservationUpdate = catchAsyncError(async (req, res, next) => {
  const reserver = await Reservation.findById(req?.params?.id);
  console.log("ğŸš€ ~ reservationUpdate ~ reserver:", reserver);

  if (!reserver) {
    return next(new ErrorHandler("Reserver bulunamadÄ± !", 404));
  }

  if (reserver.status === "Ã‡Ä±kÄ±ÅŸ YaptÄ±") {
    return next(new ErrorHandler("Bu kiÅŸi zaten Ã§Ä±kÄ±ÅŸ yaptÄ±. !", 404));
  }
  console.log(req.body.status);
  if (req.body.status === "Ã‡Ä±kÄ±ÅŸ YaptÄ±") {
    reserver.status = "Ã‡Ä±kÄ±ÅŸ YaptÄ±";
    reserver.times = "";
  } else {
    reserver.status = req.body.status;
  }
  await reserver.save();
  res.status(200).json({
    success: true,
    message: "Rezervasyon durumu baÅŸarÄ±yla gÃ¼ncellendi.",
  });
});

export default { reservationSave, reservationGet, reservationUpdate };
